src/bindings.rs

```diff
@@ -64,13 +84,23 @@ impl Preprocessor {
         let result = self.core.process(
             &src,
             Options {
-                filename: filename.map(|f| f.into())
-            }
+                filename: filename.map(|f| f.into()),
+            },
         );
 
         match result {
             Ok(output) => Ok(output),
             Err(err) => Err(as_javascript_error(err, self.core.source_map()).into()),
         }
     }
+
+    pub fn parse(&self, src: String, filename: Option<String>) -> Result<JsValue, JsValue> {
+        let result = self
+            .core
+            .parse(&src)
+            .map_err(|err| js_error("parse error".into()))?;
+        let serialized =
+            serde_json::to_string(&result).map_err(|err| js_error("serialization error".into()))?;
```

Comment:
This serialization error is more or less an assertion, but we should probably say something like "if you encounter this error, please open an issue"